client/state/analytics/actions.js

```diff
@@ -0,0 +1,47 @@
+import fromPairs from 'lodash/fromPairs';
+
+import {
+	BUMP_STAT,
+	RECORD_EVENT,
+	RECORD_PAGE_VIEW
+} from './constants.js';
+
+export const withAnalytics = ( analytics, action ) =>
```

Comment:
Is that designed to wrap a generic actionCreator and add 'stat-bumping' abilities to a fired action?
I would like to see usage of if in `mapActionsToProps` :)

Also, as a sidenote:
Please, keep in mind that actionCreators can be `thunks` in which case this will fail miserably :)